plasmapy/atomic/tests/test_particle_class.py

```diff
@@ -524,3 +565,32 @@ def test_particle_is_electron(p, is_one):
 def test_particle_bool_error():
     with pytest.raises(AtomicError):
         bool(Particle('e-'))
+
+
+def test_particle_inversion():
+    """
+    Test that the unary operator finds the antiparticle of all special
+    particles and antiparticles.
+    """
+
+    for particle_symb in ParticleZoo.everything:
+
+        particle = Particle(particle_symb)
+
+        try:
+            opposite = ~particle
+        except Exception:
+            raise InvalidParticleError(
+                "The unary operator is unable to find the antiparticle "
+                f"of {particle}.")
+
+        assert particle.integer_charge == -opposite.integer_charge, \
+            (f"The charges of {particle} and {opposite} are not "
+             f"opposites, which indicates that there is a problem "
+             f"using the unary operator to find the antiparticle of "
+             f"{particle}.")
+
+        assert particle._attributes['mass'] == opposite._attributes['mass'], \
+            (f"The masses of {particle} and {opposite} are not equal, "
+             f"which indicates that there is a problem using the unary "
+             f"operator to find the antiparticle of {particle}.")
```

Comment:
Well documented testing right here. However, could we try a non-sequential approach? I'll PR a proposal into your branch!

Reply:
Aaaand namurphy/PlasmaPy#1 is live!

Reply:
My original intention with this test was only to check whether or not the `~` returned the antiparticle, so with that in mind, I used sequential tests on purpose to avoid getting redundant error messages.  I thought about this more while cooking just now because turnips are boring, and came to the conclusion that I was thinking too narrowly and that it is worthwhile to rigorously check all of the properties too, so your PR will be a good basis for that.